linter/lints/common.go

```diff
@@ -36,12 +38,22 @@ func GetExtWithOID(exts []pkix.Extension, oid asn1.ObjectIdentifier) *pkix.Exten
 	return nil
 }
 
-// ReadASN1BooleanWithTag reports whether the incoming bytes from a decoded
-// ASN.1 Tag are a valid X.690 BOOLEAN. The caller is responsible for parsing
-// the ASN.1 Tag.
-func ReadASN1BooleanWithTag(parsedByes []byte) bool {
+// ReadOptionalASN1BooleanWithTag attempts to read the contents of an optional
+// DER-encoded ASN.1 element tagged with the given tag from incoming. It reports
+// whether the read was successful and the value of the boolean.
+func ReadOptionalASN1BooleanWithTag(incoming *cryptobyte.String, tag cryptobyte_asn1.Tag) (ok bool, present bool) {
```

Comment:
Per our discussion just now the function signature here should match `ReadOptionalASN1Boolean`:

```
ReadOptionalASN1BooleanWithTag(incoming *cryptobyte.String, out *bool, tag asn1.Tag, defaultValue bool) bool
```

Nicely, this matches the proposal in https://github.com/golang/go/issues/43019#issuecomment-1337152641